service-jehlomat/src/main/kotlin/api/SyringeApi.kt

```diff
@@ -34,6 +34,42 @@ fun Route.syringeApi(database: DatabaseService, mailer: MailerService): Route {
             call.respond(HttpStatusCode.OK, SyringeFilterResponse(filteredSyringes, pageInfo))
         }
 
+        authenticate(JWT_CONFIG_NAME) {
+
+            post("export") {
+                val loggedInUser = jwtManager.getLoggedInUser(call, database)
+                val user = database.selectUserById(loggedInUser.id)
+                val organization = user?.let { it1 -> database.selectOrganizationById(it1.organizationId) }
+                val roles = PermissionService.determineRoles(loggedInUser, organization)
```

Comment:
You don't need to fetch the organization object to find out whether the user is the admin for their organization. You can use PermissionService.determineRoles(loggedInUser, loggedInUser) for it.

I agree that it is not straightforward construction, we could create some overload variant like PermissionService.determineRoles(loggedInUser)

Reply:
c9ac85de636fc91d199464d219f9de030c7cdf37